backend/src/main/java/fk/prof/backend/model/aggregation/impl/AggregationWindowLookupStoreImpl.java

```diff
@@ -0,0 +1,42 @@
+package fk.prof.backend.model.aggregation.impl;
+
+import com.google.common.base.Preconditions;
+import fk.prof.backend.aggregator.AggregationWindow;
+import fk.prof.backend.model.aggregation.AggregationWindowLookupStore;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class AggregationWindowLookupStoreImpl implements AggregationWindowLookupStore {
+  private Map<Long, AggregationWindow> windowLookup = new ConcurrentHashMap<>();
+
+  //NOTE: Called on http event loop
+  @Override
+  public AggregationWindow getAssociatedAggregationWindow(long workId) {
+    return this.windowLookup.get(workId);
+  }
+
+  //NOTE: Called always by backend daemon thread and not on http event loop
+  @Override
+  public void associateAggregationWindow(long[] workIds, AggregationWindow aggregationWindow)
+      throws IllegalStateException {
+    Preconditions.checkNotNull(aggregationWindow);
```

Comment:
No, then NPE will directly map to bad param value. This is throwing NPE anyway, except with additional code to be executed.

We get the same benefits with a final param if it is de-referenced in the same frame.